src/main/java/com/aws/greengrass/deployment/DeploymentService.java

```diff
@@ -498,12 +487,98 @@ private void cancelCurrentDeployment() {
                 } else {
                     logger.atInfo().kv(DEPLOYMENT_ID_LOG_KEY_NAME, currentDeploymentTaskMetadata.getDeploymentId())
                             .log("Deployment is in a stage where it cannot be cancelled,"
-                                         + " need to wait for it to finish");
+                                    + " need to wait for it to finish");
                 }
             }
         }
     }
 
+    @SuppressWarnings("PMD.NullAssignment")
+    private void finishCurrentDeployment() throws InterruptedException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.